warehouse/forklift/legacy.py

```diff
@@ -767,6 +768,21 @@ def _is_duplicate_file(db_session, filename, hashes):
     return None
 
 
+def extract_wheel_metadata(path):
+    """
+    Extract METADATA file and return it as a content. The name of the
+    .whl file is used to find the corresponding .dist-info dir.
+
+    See https://www.python.org/dev/peps/pep-0658/#specification
+    """
+    global _wheel_file_re
```

Comment:
does not need this

Reply:
REs are precompiled on load, and therefore global.
This one is also used more than once.
And explicit is better than implicit.

Reply:
> And explicit is better than implicit.

`global` isn't for such purpose, it is needed only when you need to reassign the value.

Reply:
effectively. 

which is not done in this function, hence my comment @abitrolly

you do not do : 

```python

import some_module

def some_function(...):
    global some_module
    some_module.foobar()
```

for instance ;)

the less globals/`global` the better.